Heimdall/OAuthAuthorizationGrant.swift

```diff
@@ -14,12 +14,19 @@ public enum OAuthAuthorizationGrant {
     /// :param: refreshToken The refresh token.
     case RefreshToken(String)
 
+    /// An extension grant
+    ///
+    /// :param: grantType the grant type URI of the extension grant
+    /// :param: a dictionary of parameters
+    case Extension(String, [String: String])
+
     /// Returns the grant's parameters.
     ///
     /// Except for `grant_type`, parameters are specific to each grant:
     ///
     /// - `.ResourceOwnerPasswordCredentials`: `username`, `password`
     /// - `.Refresh`: `refresh_token`
+    /// - `.Extension`: `grantType`, `parameters`
```

Comment:
My intention was to describe the parameters set in the returned array, e.g., `.Refresh` sets `grant_type` and `refresh_token`. So maybe something like: `Any parameters set in the extension's parameters array`.